grid-proxy/internal/gpuindexer/gpuindexer.go

```diff
@@ -60,13 +68,67 @@ func NewNodeGPUIndexer(
 	return indexer, nil
 }
 
+// startDBListener sets up a PostgreSQL listener to listen for changes in the database and triggers the nodesChangeChan channel.
+func (n *NodeGPUIndexer) startDBListener(ctx context.Context, psqlInfo string) {
+	listener := pq.NewListener(psqlInfo, minListenerReconnectInterval, 6*minListenerReconnectInterval, func(ev pq.ListenerEventType, err error) {
+		if err != nil {
+			log.Error().Err(err).Msg("failed listening to DB changes")
+		}
+	})
+	defer listener.Close()
+
+	err := listener.Listen(dbNotificationChannel)
+	if err != nil {
+		log.Error().Err(err).Msg("failed to listen to DB changes")
+		return
+	}
+
+	for {
+		select {
+		case notification, ok := <-listener.Notify:
+			if !ok {
+				log.Error().Msg("DB listener channel closed")
+				return
+			}
+			if notification == nil {
+				log.Error().Msg("received nil notification from DB listener")
+				continue
+			}
+
+			payload := notification.Extra
+			twinId, err := strconv.ParseInt(payload, 10, 64)
+			if err != nil {
+				log.Error().Err(err).Msgf("failed to parse twin id %v", payload)
+				continue
+			}
+
+			log.Debug().Msgf("Received data from channel [%v]: twin_id: %v", notification.Channel, payload)
+
+			n.nodesChangeChan <- twinId
+		case <-ctx.Done():
+			log.Error().Err(ctx.Err()).Msg("context canceled")
+			return
+		}
+	}
+}
+
+func (n *NodeGPUIndexer) getGPUInfo(ctx context.Context, twinId int64) {
```

Comment:
twinId should uint32, and do the type conversion before passing to the function

Reply:
found it `uint64` for the most of the types, so edited it to be `uint64` instead